tests/test_upload.py

```diff
@@ -285,3 +287,20 @@ def none_upload(*args, **settings_kwargs):
     assert "pypipassword" == upload_settings.password
     assert "pypiuser" == upload_settings.username
     assert "/foo/bar.crt" == upload_settings.cacert
+
+
+def test_check_status_code(make_settings, capsys):
+    upload_settings = make_settings()
+
+    # override default upload_settings
+    upload_settings.repository_config['repository'] = \
+        "https://upload.pypi.org"
```

Comment:
You could get some bonus points for using `pytest.mark.parametrize` to test multiple URLs. ;)

Reply:
Absolutely random URLs (with pypi.org still in them) will not work as 404s are caught first. I'm adding `https://upload.pypi.org/` and `https://test.pypi.org/`.

Reply:
Added in `d2cc71d`.

Reply:
Nice. Good call on the random URLs; I was thinking of `https://pypi.org`.

Reply:
Added in 9068268.